frontend/src/components/object/ObjectPermission.vue

```diff
@@ -0,0 +1,172 @@
+      current-page-report-template="{first}-{last} of {totalRecords}"
+      :rows-per-page-options="[10, 20, 50]"
+    >
+      <template #empty>
+        <div class="flex justify-content-center">
+          <h3>There are no users associated with this object.</h3>
+        </div>
+      </template>
+      <Column
+        field="fullName"
+        header="Name"
+      />
+      <Column
+        header="Read"
+        body-class="content-center"
+      >
+        <template #body="{ data }">
+          <Checkbox
+            v-model="data.read"
+            input-id="read"
+            :binary="true"
+            @input="(value) => updateObjectPermission(value, data.userId, Permissions.READ)"
+          />
+        </template>
+      </Column>
+      <Column
+        header="Update"
+        body-class="content-center"
+      >
+        <template #body="{ data }">
+          <Checkbox
+            v-model="data.update"
+            input-id="update"
+            :binary="true"
+            @input="(value) => updateObjectPermission(value, data.userId, Permissions.UPDATE)"
+          />
+        </template>
+      </Column>
+      <Column
+        header="Delete"
+        body-class="content-center"
+      >
+        <template #body="{ data }">
+          <Checkbox
+            v-model="data.delete"
+            input-id="delete"
+            :binary="true"
+            @input="(value) => updateObjectPermission(value, data.userId, Permissions.DELETE)"
+          />
+        </template>
+      </Column>
+      <Column
+        header="Manage"
+        body-class="content-center"
+      >
+        <template #body="{ data }">
+          <Checkbox
+            v-model="data.manage"
+            input-id="manage"
+            :binary="true"
+            :disabled="!data.elevatedRights"
+            @input="(value) => updateObjectPermission(value, data.userId, Permissions.MANAGE)"
+          />
+        </template>
+      </Column>
+      <Column header="Remove">
+        <template #body="{ data }">
+          <Button
+            class="p-button-lg p-button-text"
+            style="color: red"
+            @click="removeBucketUser(data.userId)"
+          >
+            <font-awesome-icon icon="fa-solid fa-user-xmark" />
+          </Button>
+        </template>
+      </Column>
+    </DataTable>
+  </div>
+</template>
+
+<style lang="scss" scoped>
+:deep(.content-center) {
+  text-align: center !important;
+}
+
+:deep(.p-datatable-thead > tr > th) {
+  background-color: transparent;
+}
+
+:deep(.p-column-title) {
+  font-weight: bold;
+}
+
+:deep(.p-paginator) {
+  justify-content: right;
+}
+
+:deep(.header-right .p-column-header-content) {
+  justify-content: right;
+}
+
+:deep(.content-right) {
+  text-align: right !important;
+}
+
+:deep(.p-button.p-button-lg) {
+  padding: 0;
+  margin-left: 1rem;
+}
+
+:deep(.truncate) {
+  max-width: 1px;
+  white-space: nowrap;
+
+  > div {
+    overflow: hidden;
+    text-overflow: ellipsis;
+  }
+}
+</style>
```

Comment:
This CSS is the same as the bucket one. Possibly this whole table component could be generalized...
At the least extract the CSS at some point